test/libdnf/rpm/test_package.cpp

```diff
@@ -0,0 +1,346 @@
+    const auto ipr = get_pkg("npm-1:5.12.1-1.fc29.x86_64").get_prereq_ignoreinst();
+    // TODO no prereq_ignoreinst (called Ipr: in libsolv testcases) in test packages
+    const std::vector<std::string> expected = {};
+
+    CPPUNIT_ASSERT_EQUAL(expected, to_vector(ipr));
+}
+
+
+void RpmPackageTest::test_get_regular_requires() {
+    const auto reg_reqs = get_pkg("filesystem-3.9-2.fc29.x86_64").get_regular_requires();
+    // TODO doesn't test a difference between requires() and this, since there are no prereqs
+    const std::vector<std::string> expected = {"setup"};
+
+    CPPUNIT_ASSERT_EQUAL(expected, to_vector(reg_reqs));
+}
+
+
+void RpmPackageTest::test_get_recommends() {
+    const auto recommends = get_pkg("abcde-2.9.2-1.fc29.noarch").get_recommends();
+    const std::vector<std::string> expected = {"flac"};
+
+    CPPUNIT_ASSERT_EQUAL(expected, to_vector(recommends));
+}
+
+
+void RpmPackageTest::test_get_suggests() {
+    const auto suggests = get_pkg("abcde-2.9.2-1.fc29.noarch").get_suggests();
+    const std::vector<std::string> expected = {"lame"};
+
+    CPPUNIT_ASSERT_EQUAL(expected, to_vector(suggests));
+}
+
+
+void RpmPackageTest::test_get_enhances() {
+    const auto enhances = get_pkg("abcde-2.9.2-1.fc29.noarch").get_enhances();
+    // TODO no enhances in test packages
+    const std::vector<std::string> expected = {};
+
+    CPPUNIT_ASSERT_EQUAL(expected, to_vector(enhances));
+}
+
+
+void RpmPackageTest::test_get_supplements() {
+    const auto supplements = get_pkg("glibc-langpack-af-2.28-9.fc29.x86_64").get_supplements();
+    const std::vector<std::string> expected = {"(glibc and (langpacks-af or langpacks-af_ZA))"};
+
+    CPPUNIT_ASSERT_EQUAL(expected, to_vector(supplements));
+}
+
+
+void RpmPackageTest::test_get_baseurl() {
+    CPPUNIT_ASSERT_EQUAL(std::string(), get_pkg("CQRlib-1.1.1-4.fc29.x86_64").get_baseurl());
+}
+
+
+void RpmPackageTest::test_get_location() {
+    CPPUNIT_ASSERT_EQUAL(
+        std::string("x86_64/CQRlib-1.1.1-4.fc29.x86_64.rpm"),
+        get_pkg("CQRlib-1.1.1-4.fc29.x86_64").get_location()
+    );
+}
+
+
+void RpmPackageTest::test_get_checksum() {
+    CPPUNIT_ASSERT_EQUAL(
+        std::string("8b1968310a8409bbca2c06d0217b010e83a82fbb515627d87c52ef7ba084e8b3"),
+        get_pkg("CQRlib-1.1.1-4.fc29.x86_64").get_checksum().get_checksum()
+    );
+}
+
+
+void RpmPackageTest::test_get_hdr_checksum() {
+    // TODO is it correct it's empty?
+    CPPUNIT_ASSERT_EQUAL(std::string(""), get_pkg("CQRlib-1.1.1-4.fc29.x86_64").get_hdr_checksum().get_checksum());
+}
+
+
+void RpmPackageTest::test_is_installed() {
+    // TODO implement installing packages for tests
+    CPPUNIT_ASSERT_EQUAL(false, get_pkg("CQRlib-1.1.1-4.fc29.x86_64").is_installed());
+}
+
+
+void RpmPackageTest::test_get_hdr_end() {
+    // TODO what is this API method for? do we really want it?
+    CPPUNIT_ASSERT_EQUAL(6208llu, get_pkg("CQRlib-1.1.1-4.fc29.x86_64").get_hdr_end());
```

Comment:
Please could you investigate it in dnf4/libdnf?

Reply:
It is on the API, but unused. It's hard to imagine any API user needing that...

Reply:
It was part of python API, therefore we cannot say it is unused.

Reply:
We cannot say that, but we can make an educated guess the information obtained from the method is almost certainly useless to dnf API users and drop the method. On the rare occasion it would be needed, we can always add it on request.

Reply:
I discussed the functionality with dmach and we agree that there are several cases when the function/information could be useful. Please could you remove TODO?

Reply:
Sure. Out of curiosity, what are the usecases?

Reply:
You can download only rpm header, or only payload when you know where header ends.